### ISSUE TITLE
Failing to install guile 3.0.7 on Ubuntu 20.04 Docker Image

### ISSUE THREAD

**Issue by drujensen** [indiebrain/asdf-guile#4]
When attempting to install guile version 3.0.7 on a base image of ubuntu 20.04 using a `Dockerfile`, 

I get the following error:
```
=> RUN asdf plugin-add guile https://github.com/indiebrain/asdf-guile.git
=> RUN asdf install guile
------
 > [34/82] RUN asdf install guile:
#39 0.316 Downloading https://ftp.gnu.org/gnu/guile/guile-3.0.7.tar.xz
######################################################################## 100.0%         
#39 2.279 Downloading https://ftp.gnu.org/gnu/guile/guile-3.0.7.tar.xz.sig
######################################################################## 100.0%         
#39 2.593 Verifying guile-3.0.7.tar.xz
#39 2.600 gpg: directory '/root/.gnupg' created
#39 2.600 gpg: keybox '/root/.gnupg/pubring.kbx' created
#39 2.601 gpg: assuming signed data in '/tmp/asdf_guile_hqUvuN/guile-3.0.7.tar.xz'
#39 2.664 gpg: Signature made Mon May 10 08:47:33 2021 UTC
#39 2.664 gpg:                using RSA key 4FD4D288D445934E0A14F9A5A8803732E4436885
#39 2.680 gpg: requesting key A8803732E4436885 from hkp server keys.gnupg.net
#39 2.698 gpg: Can't check signature: No public key
------
executor failed running [/bin/sh -c asdf install guile]: exit code: 1
```

I've tried other versions as well.

**Comment by kozmikyak** [comment]
@drujensen : (and the author of this ASDF plugin):


I found from perusing the net that the public GPG keyserver at keys.gnupg.net was shut down some time ago.  Its keys have been published elsewhere.

I was able to get an "asdf install guile latest" command to work by importing the public key from a different keyserver:

`
gpg --keyserver keyserver.ubuntu.com --recv-keys 4FD4D288D445934E0A14F9A5A8803732E4436885
`
That, as a one-time command run form the account using ASDF, would let further runs use the cached copy of the local key without changes to this plugin's code.

I would bet that by changing the keyserver in ~/.asdf/plugins/guile/bin/install, these lines:
`
    $gpg_cmd --keyserver hkp://keys.gnupg.net \
             --keyserver-options auto-key-retrieve \
             --verify "$signature_path"
`

to
`
    $gpg_cmd --keyserver hkp://keyserver.ubuntu.com \
             --keyserver-options auto-key-retrieve \
             --verify "$signature_path"
`

it might start working without the need to cache a local copy of the public key.  I don't feel qualified to do that even as a pull request, and I don't know if the public key was different at some time in the past for old versions of Guile, and might not be on keyserver.ubuntu.com.  I see no discussion on the Guile mailing list about it.

See verification of its deprecation here:  [https://dev.gnupg.org/T5527#:~:text=Open%2C%20LowPublic,Actions](https://dev.gnupg.org/T5527#:~:text=Open%2C%20LowPublic,Actions)

See recommendations of other keyservers here:

[https://alternativeto.net/software/openpgp-keyserver-by-b4ckbone-de-/](https://alternativeto.net/software/openpgp-keyserver-by-b4ckbone-de-/)

**Comment by kozmikyak** [comment]
@drujensen:
I've verified that changing that line to refer to hkp://keyserver.ubuntu.com seems to work, after removing the local cache of the temporary workaround public key.

I don't know if it works for ancient versions of Guile.  I will do some testing and let you know what I find.  If my solution works, I'm willing to rework it as a pull request.

You can find my test repo with this change in it at 

https://github.com/kozmikyak/asdf-guile

You can install this version of the Guile plugin with

asdf plugin add guile https://github.com/kozmikyak/asdf-guile

Then try 

asdf install guile latest

as a baseline to see if it works for 3.0.8, and then try some other versions and give me some feedback if you like, in my repo so as to not scribble all over this one.  I'm glad to contribute this one-line change if it seems to work, and I'm glad for my repo to be usable as a temp workaround if it gets people running right now, but I'm not really interested in maintaining a fork over time.

**Comment by indiebrain** [comment]
Thanks for the very clear description of the issue @drujensen and @kozmikyak and for shring your findings. Fixed by: 168a88a. `asdf plugin update guile` should pull in the keyserver changes. 

FWIW, I opted to use the official `hkp://keys.openpgp.org` keyserver instead of the Ubuntu specific one proposed in this issue. Seems like the Ubuntu host is probably a mirror of the openpgp keyserver anyway? If y'all have more context on the Ubuntu keyserver and why it is a superior choice I'd be glad of edification. 

Apologies for the insanely slow response on this, the notifications of this issue fell through the cracks and I recently noticed that regular CI builds haven't been running for this project for some time. I'm attempting to remedy both problems.
